scripts/migrate.py

```diff
@@ -315,13 +308,21 @@ def deep_update(dict1, dict2):
                           {
                               'branch': addon_config.get('branch', 'master'),
                               'url': addon_config['url'],
-                              'target': os.path.join(options.branch_dir,
-                                                     version,
-                                                     name),
+                              'target': target,
                           })
             else:
                 raise Exception('Unknown type %s' % addon_config_type)
 
+        realtarget = os.path.realpath(target)
+        for root, dirs, files in os.walk(realtarget):
+            for file in files:
+                if file == "__openerp__.py":
+                    linkpath = os.path.join(options.branch_dir,
+                                            version, 'addons', name)
+                    if os.path.exists(linkpath):
+                        os.remove(linkpath)
+                    os.symlink(root, linkpath)
```

Comment:
Yes, exists a best optimized code. I will work on it, then commit it to this  pull request.